test/azul_test_case.py

```diff
@@ -344,3 +347,18 @@ def setUp(self):
                 with self.assertRaises(KeyboardInterrupt):
                     TC(events).run()
                 self.assertEqual(events, expected_events)
+
+
+class AzulMotoTestCase(TestCase):
+    """
+    A mixin for tests that utilize the moto module
+    """
+
+    def setUp(self) -> None:
+        super().setUp()
+        # Moto backends are reset to ensure no residual resources are left over if
+        # a test fails to cleanup after itself.
+        moto_api_backend.reset()
+
+    def tearDown(self) -> None:
+        super().tearDown()
```

Comment:
This seems unnecessary

Reply:
> You reference the issue in two different commit titles. I don't believe this is correct. I would remove it from "Add super setup/teardown to VersionTableTestCase" since this commit doesn't directly address the underlying issue.

Thanks for catching that, I had first added the moto backend resets as a part of the teardown for the AzulUnitTestCase, so it was necessary for the fix. I've updated the commit titles.